sympy/polys/domains/mpelements.py

```diff
@@ -22,10 +22,7 @@ def _set_mpf(self, val):
         prec, rounding = self.context._prec_rounding
         tol = self.context.tol
 
-        if mpf_lt(mpf_abs(val, prec, rounding), tol):
-            self.__mpf__ = fzero
-        else:
-            self.__mpf__ = val
+        self.__mpf__ = val
```

Comment:
The docstring at the top of the file says "Real and complex elements with built-in truncation." My only fear is that although the change you have made fixes an issue, there might be a lot more to give attention to: 1) the unused `prec`, `rounding` and `tol` are orphaned variables now (that should probably be deleted); 2) the docstring at the top of the file should reflect what is actually happening in the file; and 3) perhaps the same changes should be applied to the lines at 39 and following (which deal with complex numbers). 

I'm just not sure that the change is commensurate with the amount of thought that needs to go into making sure that this is what is intended by this file's purpose.

> It holds ~15 digits. You do not want this to become 0, only because the number is smaller than 1e-16

Yes, I think you are right here. Should *any* sort of truncation (perhaps in digits) be happening in this file?